src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpPreferIsExactSpellingIsTrueForKnownApis.Fixer.cs

```diff
@@ -0,0 +1,164 @@
+            }
+            else
+            {
+                editor.ReplaceNode(possibleExistingArgument, replacement);
+            }
+        }
+
+        private static async Task<Document> ApplixFixAndGetChangedDocument(DocumentEditor editor, Task fixingTask)
+        {
+            await fixingTask.ConfigureAwait(false);
+            return editor.GetChangedDocument();
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            if (!model.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDllImportAttribute, out var dllImportType))
+            {
+                return;
+            }
+
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            if (root.FindNode(context.Span) is not MethodDeclarationSyntax methodDeclaration)
+            {
+                return;
+            }
+
+            if (model.GetDeclaredSymbol(methodDeclaration, context.CancellationToken) is not IMethodSymbol methodSymbol)
+            {
+                return;
+            }
+
+            var dllImportAttribute = methodSymbol
+                .GetAttributes()
+                .FirstOrDefault(x => x.AttributeClass.Equals(dllImportType));
+            if (dllImportAttribute is null)
+            {
+                return;
+            }
+
+            var dllName = dllImportAttribute.ConstructorArguments[0].Value.ToString();
+            string title = MicrosoftNetCoreAnalyzersResources.PreferIsExactSpellingIsTrueForKnownApisTitle;
+
+            var editor = await DocumentEditor.CreateAsync(context.Document, context.CancellationToken).ConfigureAwait(false);
+            if (!PreferIsExactSpellingIsTrueForKnownApisAnalyzer.KnownApis.Value.TryGetValue(dllName, out var methods))
+            {
+                return;
+            }
+
+            var actualName = methodSymbol.GetActualExternName(dllImportAttribute);
+            if (methods.Contains(actualName))
+            {
+                if (actualName.EndsWith("W", StringComparison.OrdinalIgnoreCase))
+                {
+                    context.RegisterCodeFix(new PreferIsExactSpellingChangeAction(title,
+                                                     async ct => await ApplixFixAndGetChangedDocument(editor, AddExactSpelling(context.Document, methodDeclaration, editor, ct)).ConfigureAwait(false),
+                                                     equivalenceKey: title),
+                                    context.Diagnostics);
+                    return;
+                }
+                if (actualName.EndsWith("A", StringComparison.OrdinalIgnoreCase))
+                {
+                    context.RegisterCodeFix(new PreferIsExactSpellingChangeAction(title,
```

Comment:
I don't see any difference with the previous if block, if that's intended, you could merge the 2.